### DISCUSSION TITLE
URL structure is determined in the backend. can It be handled with react-router-7 properly?

### DISCUSSION THREAD

**Discussion by pooya-badiee** [remix-run/react-router#12526] [Q&A] [answered]
# Handling Dynamic URL Rewrites in Frontend

## Current Setup
Our site implements a custom URL rewrite system. For example, a route like `/special-tablet/new` requires a two-step process:

1. First, we make a backend API call to resolve the custom route and determine its target page type
2. Then, based on the response (e.g., "product page"), we load the appropriate content

![image](https://github.com/user-attachments/assets/a20ceaa8-af31-4869-93fc-23ff25b87408)

What's the recommended architectural approach for handling dynamic URL rewrites in react-router framework mode?
we are currently using react-router-5 and I want to see if there is a possibility to upgrade to the framework mode.
Any examples or best practices would be appreciated.

Thank you

**Comment by sergiodxa** [comment, marked as answer]
You can use a [catch-all/splat route](https://reactrouter.com/start/framework/routing#splats) and inside the route loader fetch the API to know how to handle that route.

Although at that point you're not using RR anymore, you're just using the API as your router, unless you other routes not handle by the API like `/`, `/login`, etc. in that case it can make sense to use RR to define those routes and a catch-all/splat for the rest.
